ansible_collections/arista/avd/plugins/filter/convert_dicts.py

```diff
@@ -0,0 +1,70 @@
+#
+# def arista.avd.convert_dicts
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from jinja2.runtime import Undefined
+
+
+def convert_dicts(dictionary, primary_key="name"):
+    """
+    The `arista.avd.convert_dicts` filter will convert a dictionary containing nested dictionaries to a list of dictionaries
+    and insert the outer dictionary keys into each list item using the primary_key `name` (key name is configurable).
+
+    This filter is intended for:
+
+    - Seemless data model migration from dictionaries to lists.
+    - Improve Ansible's processing performance when dealing with large dictionaries by converting them to lists of dictionaries.
+
+    Note: If the variable is already a list, it will pass through untouched
+
+    To use this filter:
+
+    ```jinja
+    {# convert list of dictionary with default `name:` as the primary key #}
+    {% set example_list = example_dictionary | arista.avd.convert_dicts %}
+    {% for example_item in example_list %}
+    item primary key is {{ example_item.name }}
+    {% endfor %}
+
+    {# convert list of dictionary with `id:` set as the primary key #}
+    {% set example_list = example_dictionary | arista.avd.convert_dicts('id') %}
+    {% for example_item in example_list %}
+    item primary key is {{ example_item.id }}
+    {% endfor %}
+    ```
+
+    Parameters
+    ----------
+    dictionary : any
+        Nested Dictionary to convert - returned untouched if not a nested dictionary
+    primary_key : str, optional
+        Name of primary key used when inserting outer dictionary keys into items.
+
+    Returns
+    -------
+    any
+        Returns list of dictionaries or input variable untouched if not a nested dictionary
+    """
+    if not isinstance(dictionary, dict):
+        # Not a dictionary, return the original
+        return dictionary
+    else:
+        output = []
+        for key in dictionary.keys():
```

Comment:
I saw this article online: https://stackoverflow.com/questions/44244281/how-come-key-in-dictionary-keys-slower-than-key-in-dictionary, which says to avoid using .keys(). 

Also I tried this python3 code my self:

```
import string, time

d = dict.fromkeys(string.ascii_lowercase, 0)

def f_nokeys(d):
    start_time = time.time()
    print("t" in d)
    print("Without .keys method")
    print("--- %s seconds ---" % (time.time() - start_time))

def f_keys(d):
    start_time = time.time()
    print("t" in d.keys())
    print("With .keys method")
    print("--- %s seconds ---" % (time.time() - start_time))

f_nokeys(d)
print('\n\n')
f_keys(d)
```

Most of the time the .keys() method is taking much more time.
```
docker % python test.py
True
Without .keys method
--- 2.4080276489257812e-05 seconds ---



True
With .keys method
--- 5.9604644775390625e-06 seconds ---
```

Reply:
I tried your function (convert to list) and found no difference with .keys() method and without it. I could not find a major difference in time.

Reply:
Just tested on my end too, and even if I don't see any measurable difference, I will add this fix (remove `.keys()`) together with the next refactor. Thank you for the deep review!